### PR TITLE
migrate [MozillaObservatory] to new API

### LINKED ISSUE #10396: MDN HTTP Observatory is launched, and Mozilla Observatory is now deprecated.
### Are you experiencing an issue with...

shields.io

### 🐞 Description

Today I noticed our [Mozilla Observatory Grade](https://shields.io/badges/mozilla-http-observatory-grade) badge was no longer working. 

After revieing the [linked page](https://observatory.mozilla.org/analyze/docs.qlcplus.org) I noticed that the service has been moved.

image

Please see link to the blog post: https://developer.mozilla.org/en-US/blog/mdn-http-observatory-launch/

> Today, we are delighted to announce that the HTTP Observatory's new home is MDN! This blog post explains a little bit of the history behind the tool, how we got to where we are now, what the migration means for Observatory and MDN, and what Observatory tests.

### 🔗 Link to the badge

https://shields.io/badges/mozilla-http-observatory-grade

### 💡 Possible Solution

I think we will need to embrace the new MDN HTTP observatory and change the badge accordingly.

https://developer.mozilla.org/en-US/observatory

### PR DESCRIPTION
**Pull request by chris48s**
Closes #10396

It seems like the old API is now back online and the badge is working again, so we don't have to merge this immediately. We can take a bit of time to wok out the details. But it is deprecated and due to be retired in September so we do need to migrate this soon-ish.

This is a bit tricky to work on as the API is currently undocumented (although we have been told we can use it and docs are coming). There are some behaviours that aren't exactly specified. I've made some assumptions. Some notes:

- The API seems to no longer explicitly issue a scan status. Dealing with a pending scan, for example, doesn't really seem to be a thing any more. I don't really have the examples to easily test what all the failure cases look like, although I did think of one obvious one. I think in lieu of better docs, we can say we either:
    - Get a scan result or
    - The validation fails and we reject the response with the generic "invalid" message
    We can always improve the error messages in future when we have docs to understand the failure cases a bit better.
- The concept of publishing (or not) the result seems to be gone.
- I guess this could be a service where the first request sometimes takes a long time, but once the first scan has happened we have a cached result. I haven't done it here, but I wonder if this would be a good place to use https://github.com/badges/shields/pull/9233
- This service had a bunch of tests which were done using mocks. I decided to bin them and switch those to testing the rendering function in unit tests in a `.spec` file

### CODE REVIEW

**Comment by yestalgia** [comment]
Thank you for your quick action on this one. It's greatly appreciated.

**Comment by nicwortel** [comment]
Thank you for creating this PR!

> It seems like the old API is now back online and the badge is working again, so we don't have to merge this immediately. We can take a bit of time to wok out the details. But it is deprecated and due to be retired in September so we do need to migrate this soon-ish.

For me it still appears to be broken:

afbeelding

**Review by PyvesB — DISMISSED** [review]
Thanks for the migration!

File: services/mozilla-observatory/mozilla-observatory.tester.js [hunk 1]
```diff
@@ -3,323 +3,21 @@ import { createServiceTester } from '../tester.js'
...
-  .try(
-    Joi.string().regex(/^[ABCDEF][+-]? \([0-9]{1,3}\/100\)$/),
-    Joi.string().allow('pending'),
-  )
+  .try(Joi.string().regex(/^[ABCDEF][+-]? \([0-9]{1,3}\/100\)$/))
   .required()
 
-t.create('request on observatory.mozilla.org')
+t.create('valid')
   .timeout(10000)
```
[the remark this answers is not in the document]
Comment (chris48s) [reply services/mozilla-observatory/mozilla-observatory.tester.js]: It is a good question. Anecdotally this new service does seem to be faster, but this probably depends a bit on how quickly the domain you're scanning (which is upstream again from mozilla) responds.

I think I'm going to go ahead and merge this as the old API seems to be down again. I've pushed a commit which removes the timeouts. Lets keep an eye on the daily test runs. If this ends up being flaky, we can put them back. If it is stable, then cool :sunglasses:

**Pull request merged**